src/temporal/IsoFields.js

```diff
@@ -777,9 +777,9 @@ class Unit extends TemporalUnit {
      * @returns {number}
      */
     addTo(temporal, periodToAdd) {
+        const added = MathUtil.safeAdd(temporal.get(WEEK_BASED_YEAR), periodToAdd);
```

Comment:
this calculation is only necessary for WEEK_BASED_YEARS, please only do the calculation inside the case